example/src/App.tsx

```diff
@@ -118,20 +115,30 @@ export default function App() {
       />
       {isCentral && (
         <>
-          <Button title="scan" onPress={scan} />
+          <Button
+            title="scan"
+            onPress={async () => {
+              await central.scan()
+            }}
+          />
           {peripheralId && (
             <Button
               title="connect"
-              onPress={() => central.connect(peripheralId)}
+              onPress={async () => await central.connect(peripheralId)}
+            />
+          )}
+          {connected && (
+            <Button
+              title="write"
+              onPress={async () => await central.sendMessage(msg)}
             />
           )}
-          {connected && <Button title="write" onPress={() => write(msg)} />}
         </>
       )}
       {isPeripheral && (
         <>
-          <Button title="advertise" onPress={advertise} />
-          <Button title="notify" onPress={() => indicate(msg)} />
+          <Button title="advertise" onPress={peripheral.advertise} />
```

Comment:
likely not an issue here, but could you do:
```suggestion
          <Button title="advertise" onPress={() => peripheral.advertise()} />
```
it fixes the this-scope.